Makefile

```diff
@@ -1,205 +1,512 @@
-	  $(PREFIX)/bin/fstar_tests.exe \
-	  $(PREFIX)/bin/fstar.exe \
-	  $(PREFIX)/share/fstar
-
-.PHONY: package
-package: all
-	+$(Q)$(MAKE) -C src/ocaml-output package
-
-# Removes everything created by `make all`. MUST NOT be used when
-# bootstrapping.
-.PHONY: clean
-clean: clean-intermediate
-	$(call msg, "CLEAN")
-	$(Q)cd $(DUNE_SNAPSHOT) && { dune clean || true ; }
-
-# Removes all .checked files and other intermediate files
-# Does not remove the object files from the dune snapshot.
-.PHONY: clean-intermediate
-clean-intermediate:
-	+$(Q)$(MAKE) -C ulib clean
-	+$(Q)$(MAKE) -C src clean
-
-# Regenerate all hints for the standard library and regression test suite
-.PHONY: hints
-hints:
-	+$(Q)OTHERFLAGS="${OTHERFLAGS} --record_hints" $(MAKE) -C ulib/
-	+$(Q)OTHERFLAGS="${OTHERFLAGS} --record_hints" $(MAKE) ci-uregressions ci-ulib-extra
-
-.PHONY: bench
-bench:
-	./bin/run_benchmark.sh
-
-# Regenerate and accept expected output tests. Should be manually
-# reviewed before checking in.
-.PHONY: output
-output:				\
-	output--examples	\
-	output--tests
-
-.PHONY: output--%
-output--%:
-	+$(Q)$(MAKE) -C $* accept
-
-# This rule is meant to mimic what the docker based CI does, but it
-# is not perfect. In particular it will not look for a diff on the
-# snapshot, nor run the build-standalone script.
-.PHONY: ci
-ci:
-	+$(Q)FSTAR_HOME=$(CURDIR) $(MAKE) ci-pre
-	+$(Q)FSTAR_HOME=$(CURDIR) $(MAKE) ci-post
-
-# This rule runs a CI job in a local container, exactly like is done for
-# CI.
-.PHONY: docker-ci
-docker-ci:
-	docker build -f .docker/standalone.Dockerfile \
-		--build-arg CI_THREADS=$(shell nproc) \
-		--build-arg FSTAR_CI_NO_GITDIFF=1 \
-		.
-
-.PHONY: ci-pre
-ci-pre: ci-rebootstrap
-
-.PHONY: ci-rebootstrap
-ci-rebootstrap:
-	+$(Q)$(MAKE) full-bootstrap FSTAR_BUILD_PROFILE=test
-
-.PHONY: ci-ocaml-test
-ci-ocaml-test:
-	+$(Q)$(MAKE) -C src ocaml-unit-tests
-
-.PHONY: ci-ulib-extra
-ci-ulib-extra:
-	+$(Q)$(MAKE) -C ulib extra
-
-.PHONY: ci-ulib-in-fsharp
-ci-ulib-in-fsharp:
-	+$(Q)$(MAKE) -C ulib ulib-in-fsharp
-
-.PHONY: ci-post
-ci-post:						\
-	ci-ulib-in-fsharp				\
-	ci-ocaml-test					\
-	ci-uregressions					\
-	$(if $(FSTAR_CI_TEST_KARAMEL),ci-karamel-test,)
-
-.PHONY: ci-uregressions
-ci-uregressions:
-	+$(Q)$(MAKE) -C src uregressions
-
-.PHONY: ci-karamel-test
-ci-karamel-test: ci-krmllib
-	+$(Q)$(MAKE) -C examples -f karamel.Makefile
-
-# krmllib needs FStar.ModifiesGen already checked, so we add the dependency on
-# ulib-extra here. This is possibly spurious and fixable by tweaking krml's makefiles.
-.PHONY: ci-krmllib
-ci-krmllib: ci-ulib-extra
-	+$(Q)OTHERFLAGS="${OTHERFLAGS} --admit_smt_queries true" $(MAKE) -C $(KRML_HOME)/krmllib
-
-# Shortcuts:
-
-.PHONY: 1 2 3
-
-1: fstar
-
-# This is a hacky rule to bootstrap the compiler, and not
-# the library, more quickly.
-2:
-	+$(Q)$(MAKE) -C src ocaml
-	+$(Q)$(MAKE) -C src/ocaml-output overlay-snapshots
-	+$(Q)$(MAKE) fstar
-
-3:
-	+$(Q)$(MAKE) 1
-	+$(Q)$(MAKE) 2
+SHELL=/bin/bash
```

Comment:
```suggestion
```

Is there a reason we need to hardcode the full path to bash here?

As you've observed, this doesn't work in nixpkgs.  More problematically, it doesn't work on NixOS either (where you'd need to write `make SHELL=bash` everytime you call make..).

Reply:
I think there's no need for bash at all actually, pushed a patch.

Reply:
Commented before seeing the edit, does something fail in `dash`? I think I was being too defensive by using bash since I had many complicated rules, but that logic has been pushed to some scripts (which do require bash, but the Makefile does not afaict).

Reply:
I didn't try to build on Debian, I was just figuring out the reason why you put the `SHELL=` there in the first place.

If you've already removed all the bashisms from the Makefile, then we're good now.